### DISCUSSION TITLE
USB Relay - No Serial Number (with Pi Zero running Listener)

### DISCUSSION THREAD

**Discussion by jethri** [josephdadams/TallyArbiter#671] [Q&A] [answered]
Hey folks.  Looking for some help with USB Relay attached to a Pi Zero W.  I’m trying to find my way but feels like I hit a wall.  I bought the 8-channel relay linked on the Relay Listener readme (“Zer” brand).  

It arrived, and upon plugging it in, no matter how I try to query it, it appears that the serial number is simply blank/null.  Also looked at the device with lsusb, and another tool, and no serial.

I might be jumping to a conclusion, but I think because of this, I get an error when running the listener that no USB relays have been initialized, and therefore the listener cannot update the relay state.

Here’s what I see when I run sudo npm start in /relay-listener:

`Detected Relays:
[
  {
    vendorId: 5824,
    productId: 1503,
    path: '/dev/hidraw0',
    serialNumber: '',
    manufacturer: 'www.dcttech.com',
    product: 'USBRelay8',
    release: 256,
    interface: 0,
    usagePage: 65280,
    usage: 1,
    serial: ''
  }
]
[Thu Mar 21 2024 01:40:50 GMT-0400 (Eastern Daylight Time)]     Error updating relay state: No USB relays have been initialized. Are you sure it is connected? `

Side-note, felt like I was struggling to even get to this point.  Maybe Pi Zero W doesn’t support node.js or npm/nvm versions required?  Here’s what I ended up with, tell me if this is a red flag:

`node -v && npm -v
v16.9.1
8.12.1`

Back to the serial issue, I tried using Josephdadams/usbrelay to manually figure something out, but no dice.

There’s another software at [https://github.com/darrylb123/usbrelay](https://github.com/darrylb123/usbrelay) and to my surprise, I was able to trigger a relay state on and back off (!) but even this tool does not indicate any serial number, and although it sort of indicates it can set the serial, it seems to only be an ID this software is looking at.

Did I get a dud?  Can I work around not having a serial number?  Maybe the path could be used…?    Is my node or npm version my downfall?

Thanks in advance, and for all your amazing work on this!

**Comment by josephdadams** [comment, marked as answer]
It’s because the product is `USBRelay8` and the code is looking for `USBRelay`

https://github.com/josephdadams/USBRelay/blob/4a9e09e628dbdf073705f08b655805b021915437/index.js#L38
